dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "PushNotificationChannel.h"
+#include "Microsoft.Windows.PushNotifications.PushNotificationChannel.g.cpp"
+#include <winrt\Windows.Networking.PushNotifications.h>
+#include <winrt\Windows.Foundation.h>
+
+namespace winrt::Windows
+{
+    using namespace winrt::Windows::Networking::PushNotifications;
+    using namespace winrt::Windows::Foundation;
+}
+namespace winrt::Microsoft
+{
+    using namespace winrt::Microsoft::Windows::PushNotifications;
+}
+
+namespace winrt::Microsoft::Windows::PushNotifications::implementation
+{
+    PushNotificationChannel::PushNotificationChannel(winrt::Windows::PushNotificationChannel const& channel): m_channel(channel) {}
+
+    winrt::Windows::Uri PushNotificationChannel::Uri()
+    {
+        return winrt::Windows::Uri{ m_channel.Uri() };
+    }
+    winrt::Windows::DateTime PushNotificationChannel::ExpirationTime()
+    {
+        return m_channel.ExpirationTime();
+    }
+    void PushNotificationChannel::Close()
+    {
+        m_channel.Close();
```

Comment:
This is no fail or can it throw?

Reply:
Yes it can throw. Adding a try/catch here.